ddtrace/propagation/http.py

```diff
@@ -637,18 +650,21 @@ def _extract(headers):
         # type: (Dict[str, str]) -> Optional[Context]
 
         try:
-            tp = _extract_header_value(_POSSIBLE_HTTP_HEADER_TRACEPARENT, headers)
+            tp = _TraceContext._extract_header_value_tracecontext(_POSSIBLE_HTTP_HEADER_TRACEPARENT, headers)
             if tp is None:
                 log.debug("no traceparent header")
                 return None
+            # uppercase char in tp makes it invalid
+            if tp.isupper():
```

Comment:
this will only be true if all characters are uppercase, not if a single character is uppercase.

there has to be a better way, but `if any(c.isupper() for c in tp):` would work.

(we might want a generator expression here (what I have) instead of a list comprehension to ensure we don't process any more characters than needed to find the first uppercase)

Reply:
can we just `tp = tp.lower()`? does it matter?

Reply:
Would `if not tp.islower()` be better? "A string is lowercase if all cased characters in the string are lowercase and there is at least one cased character in the string"

Reply:
We can't accept it if there is an uppercase value. I asked Zach Montoya the same question about just lowering it and moving on.